File: CWM.UI/cwm_desktop_mobile/lib/models/vehicle_service_history.dart
import 'package:cwm_desktop_mobile/models/employee.dart';
import 'package:cwm_desktop_mobile/models/enums/service.dart';
import 'package:cwm_desktop_mobile/models/user.dart';
import 'package:cwm_desktop_mobile/models/vehicle.dart';
import 'package:json_annotation/json_annotation.dart';

part 'vehicle_service_history.g.dart';

@JsonSerializable()
class VehicleServiceHistory {
  int id;
  DateTime serviceDate;
  Service serviceType;
  String description;
  String sugestions;
  Vehicle? vehicle;
  Employee? employee;
  User? user;

  VehicleServiceHistory(
      this.id,
      this.serviceDate,
      this.serviceType,
      this.description,
      this.sugestions,
      this.vehicle,
      this.employee,
      this.user);

  factory VehicleServiceHistory.fromJson(Map<String, dynamic> json) =>
      _$VehicleServiceHistoryFromJson(json);

  Map<String, dynamic> toJson() => _$VehicleServiceHistoryToJson(this);
}

File: CWM.Model/Models/VehicleServiceHistory.cs
using CWM.Core.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CWM.Core.Models
{
    public class VehicleServiceHistory : Base
    {
        public DateTime? ServiceDate { get; set; }
        public Service ServiceType { get; set; }
        public string Description { get; set; } = string.Empty;
        public string Sugestions { get; set; } = string.Empty;
        public Vehicle? Vehicle { get; set; } = new();
        public Employee? Employee { get; set; } = new();
        public User? User { get; set; } = new();
    }
}
